packages/core-mobile/app/services/network/NetworkService.ts

```diff
@@ -45,27 +45,13 @@ class NetworkService {
     }
   }
 
-  getProviderForNetwork(
+  async getProviderForNetwork(
     network: Network
-  ): JsonRpcBatchInternal | BitcoinProvider | Avalanche.JsonRpcProvider {
-    if (network.vmName === NetworkVMType.BITCOIN) {
-      return getBitcoinProvider(network.isTestnet)
-    }
-
-    if (network.vmName === NetworkVMType.EVM) {
-      return getEvmProvider(network)
-    }
-
-    if (
-      network.vmName === NetworkVMType.AVM ||
-      network.vmName === NetworkVMType.PVM
-    ) {
-      return network.isTestnet
-        ? Avalanche.JsonRpcProvider.getDefaultFujiProvider()
-        : Avalanche.JsonRpcProvider.getDefaultMainnetProvider()
-    }
-
-    throw new Error(`Unsupported network type: ${network.vmName}`)
+  ): Promise<
+    JsonRpcBatchInternal | BitcoinProvider | Avalanche.JsonRpcProvider
+  > {
+    const module = await ModuleManager.loadModuleByNetwork(network)
+    return module.getProvider(network)
```

Comment:
also did some cleaning up around the provider logic to make sure we utilize vm module for it